docs/faq.rst

```diff
@@ -50,27 +49,76 @@ Without `jit` the output is::
     Inside: 2
     Result: 4
 
-and with `jit` it is:
+and with ``jit`` it is:
 
     Inside: 0
     Result: 0
     Result: 1
     Result: 2
 
-For `jit` the function is executed once using the Python interpreter, at which time the
-`Inside` printing happens, and the first value of `y` is observed. Then the function
-is compiled and cached, and executed multiple times with different values of `x`, but
-with the same first value of `y`. 
+For ``jit``, the function is executed once using the Python interpreter, at which time the
+``Inside`` printing happens, and the first value of ``y`` is observed. Then the function
+is compiled and cached, and executed multiple times with different values of ``x``, but
+with the same first value of ```y``.
 
 Additional reading:
 
-  * [JAX - The Sharp Bits: Pure Functions](https://jax.readthedocs.io/en/latest/notebooks/Common_Gotchas_in_JAX.html#%F0%9F%94%AA-Pure-functions)
+  * `JAX - The Sharp Bits: Pure Functions <https://jax.readthedocs.io/en/latest/notebooks/Common_Gotchas_in_JAX.html#%F0%9F%94%AA-Pure-functions>`_.
+
+.. comment We refer to the anchor below in JAX error messages
+.. _value_must_be concrete:
+
+`Tracer value where concrete value is expected` error
```

Comment:
I still don't think we should set up "Tracer" and "concrete" as mutually exclusive alternatives. How about instead "Tracer where Python value is expected" as one error, and "Abstract value where concrete value is expected" as another, or similar? The former can be used for `jnp.split(x, n)` for a Tracer `n`, while the latter can be used for `jit(lambda x: x if x > 0 else 0.)(3.)` (and won't be raised on `grad(lambda x: x if x > 0 else 0.)(3.)` even though in both cases `x` refers to a Tracer instance in the body of the function being transformed).